projects/knora-ui/src/lib/viewer/values/date-value/date-value.component.ts

```diff
@@ -0,0 +1,198 @@
+import {Component, Inject, Input, OnChanges, OnDestroy, OnInit, SimpleChanges, ViewChild} from '@angular/core';
+import {CreateDateValue, KnoraDate, KnoraPeriod, ReadDateValue, UpdateDateValue} from '@knora/api';
+import {
+  AbstractControl,
+  FormBuilder,
+  FormControl,
+  FormGroup,
+  FormGroupDirective,
+  NgForm,
+  ValidatorFn
+} from '@angular/forms';
+import {Subscription} from 'rxjs';
+import {BaseValueComponent} from '../base-value.component';
+import {ErrorStateMatcher} from '@angular/material';
+import {DateInputComponent} from "./date-input/date-input.component";
+
+/** Error when invalid control is dirty, touched, or submitted. */
+export class IntervalErrorStateMatcher implements ErrorStateMatcher {
+  isErrorState(control: FormControl | null, form: FormGroupDirective | NgForm | null): boolean {
+    const isSubmitted = form && form.submitted;
+    return !!(control && control.invalid && (control.dirty || control.touched || isSubmitted));
+  }
+}
+
+@Component({
+  selector: 'kui-date-value',
+  templateUrl: './date-value.component.html',
+  styleUrls: ['./date-value.component.scss']
+})
+export class DateValueComponent extends BaseValueComponent implements OnInit, OnChanges, OnDestroy {
+
+  @ViewChild('dateInput', {static: false}) dateInputComponent: DateInputComponent;
+
+  @Input() displayValue?: ReadDateValue;
+
+  valueFormControl: FormControl;
+  commentFormControl: FormControl;
+
+  form: FormGroup;
+
+  valueChangesSubscription: Subscription;
+
+  // TODO: check that both dates have the same calendar in case of a KnoraPeriod
```

Comment:
is this `TODO` still relevant? I believe you're checking for the calendar consistency in time-input.component.ts

Reply:
done in f5d7a65